src/utils.ts

```diff
@@ -179,27 +176,53 @@ export function tokenToPythSymbol(token: string) {
   return undefined;
 }
 
-export async function getPythPrice(splMint: PublicKey) {
+export async function getPythPrice(baseMint: PublicKey) {
   const connection: Connection = new Connection(API_URL);
   const pythPublicKey = getPythProgramKeyForCluster(IS_DEV ? 'devnet': 'mainnet-beta');
   const pythClient = new PythHttpClient(connection, pythPublicKey);
   const data = await pythClient.getData();
 
   for (let symbol of data.symbols) {
     const price = data.productPrice.get(symbol)!;
-    if (tokenToPythSymbol(splMintToToken(splMint)) == symbol) {
+    if (tokenToPythSymbol(splMintToToken(baseMint)) == symbol) {
       return price.price;
     }
   }
   return 0.0;
 }
 
-export function tokenType(type: string) {
-  if (type == "Upside") {
-    return "call";
+export function getDIPType(dip_deposit: DIPDeposit) {
+  for (let premium of premiumAssets){
+    if (dip_deposit.quoteAsset == premium) {
+      return "call";
+    }
+    if (dip_deposit.baseAsset == premium) {
+      return "put";
+    }
+  }
+  return 'undefined';
+}
+
+export function getDIPDirection(dip_deposit: DIPDeposit) {
+  for (let premium of premiumAssets){
+    if (dip_deposit.quoteAsset == premium) {
+      return "Upside";
+    }
+    if (dip_deposit.baseAsset == premium) {
+      return "Downside";
+    }
   }
-  if (type == "Downside") {
-    return "put";
+  return 'undefined';
+}
+
+export function adjustStrike(dip_deposit: DIPDeposit) {
+  for (let premium of premiumAssets){
+    if (dip_deposit.quoteAsset == premium) {
+      return dip_deposit.strike;
+    }
+    if (dip_deposit.baseAsset == premium) {
+      return 1 / dip_deposit.strike;
```

Comment:
Not sure if the units work out here. Is strike based on what we use for the scalper, or what is in the DIPState on chain? The one on chain is integers only, so it is atoms of USDC per full token.
Either way, at least leave a TODO to change the property strike to include units.